packages/runtime-html/src/binding/binding-utils.ts

```
@@ -165,11 +167,62 @@ export const mixinAstEvaluator = /*@__PURE__*/(() => {
     if (vc == null) {
       throw createMappedError(ErrorNames.ast_converter_not_found, name);
     }
+    // Check for instance useCallerContext property
+    const useCallerContext = vc.useCallerContext === true;
+    // Compose caller context
+    let callerContext: any = null;
+    if (useCallerContext) {
+      // this is the binding instance
+      // .target is the element/attribute
+      // .component is the closest custom element view-model (if any)
+      let component;
+      try {
+        // Prefer to get the component from the binding's controller if available
+        const maybeController = (this as any)._controller;
+        if (maybeController instanceof Controller) {
+          component = maybeController.viewModel;
+        } else {
+          const node = (this as any).target;
+          let searchNode = node;
+          if (node && node.nodeType === 3 && node.parentNode) {
+            searchNode = node.parentNode;
+          }
+          if (searchNode && searchNode.nodeType !== undefined) {
+            const ceController = findElementControllerFor(searchNode, { optional: true });
+            component = ceController?.viewModel;
+          } else if (node && node.element) {
+            const ceController = findElementControllerFor(node.element, { optional: true });
+            component = ceController?.viewModel;
+          }
+        }
+      } catch (e) {
+        // ignore
+      }
```

why are we swallowing the error here?

---

I didn't know what kind of error to throw here or if this code would remain, so it was placeheld until ready for merge.